engine/include/cubos/engine/tools/scene_editor/plugin.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <cubos/engine/cubos.hpp>
+#include <string>
+
+namespace cubos::engine::tools
+{
+    struct SceneInfo;
+
+    /// Resource used to store Scene(s) information
+    struct SceneInfo
+    {
+        bool isExpanded{false};
+        bool isSelected{false};
+        bool shouldBeRemoved{false};
+        std::string name;
+        std::vector<std::pair<std::string, cubos::core::ecs::Entity>> entities;
+        std::vector<std::pair<std::string, SceneInfo>> scenes;
+    };
```

Comment:
Is this ever used outside of the scene editor? If not, I think it would be best to move it to the source file.